lib/rubocop/cop/lint/redundant_type_conversion.rb

```diff
@@ -52,6 +52,13 @@ module Lint
       #   {}
       #   Set.new
       #
+      #   # bad - chaining to a type constructor
+      #   Integer('25').to_i
+      #
+      #   # good - chaining to a type constructor with exceptions suppressed
+      #   # in this case, `Integer()` could return `nil`
+      #   Integer(var, exception: false).to_i
```

Comment:
I still have concerns about the compatibility description when reading it as a replacement from `bad` to `good`. How about adding an example that does not use `to_i`, like the one below? I expect that the first `good` case will be read as a general replacement that maintains compatibility.

```suggestion
      #   # bad
      #   Integer(var).to_i
      #
      #   # good
      #   Integer(var)
      #
      #   # good - chaining to a type constructor with exceptions suppressed
      #   # in this case, `Integer()` could return `nil`
      #   Integer(var, exception: false).to_i
```

Reply:
Ah yes, I see now. Good suggestion, I've updated it. :+1: